File: MidtermLabWork3/Javascript/MP09.js
const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter CSV file path: ", function(path) {

    let total = 0;
    let pass = 0;
    let fail = 0;

    try {
        const data = fs.readFileSync(path, "utf8");
        const rows = data.split("\n");

        rows.forEach(row => {
            row = row.trim();
            if (!row.startsWith('"')) return;

            const parts = row.split('",');
            if (parts.length < 2) return;

            const columns = parts[1].split(",");

            total++;
            const result = columns[6].trim().toUpperCase();

            if (result === "PASS") pass++;
            if (result === "FAIL") fail++;
        });

        console.log("\nDataset Statistics");
        console.log("--------------------");
        console.log("Total Records:", total);
        console.log("PASS:", pass);
        console.log("FAIL:", fail);

    } catch (err) {
        console.log("Error reading dataset.");
    }

    rl.close();
});

File: MidtermLabWork3/Java/MP09.java
import java.io.*;
import java.util.*;

public class MP09 {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        // Ask user for CSV dataset path
        System.out.print("Enter CSV file path: ");
        String path = scanner.nextLine();

        int total = 0;
        int pass = 0;
        int fail = 0;

        try {

            BufferedReader br = new BufferedReader(new FileReader(path));
            String line;

            while ((line = br.readLine()) != null) {
                line = line.trim();

                // Skip non-data rows
                if (!line.startsWith("\"")) continue;

                // Split name (quoted) from other columns
                String[] parts = line.split("\",", 2);
                if (parts.length < 2) continue;

                String[] data = parts[1].split(",");

                total++;

                String result = data[6].trim().toUpperCase();

                if (result.equals("PASS")) pass++;
                if (result.equals("FAIL")) fail++;
            }

            br.close();

            System.out.println("\nDataset Statistics");
            System.out.println("--------------------");
            System.out.println("Total Records: " + total);
            System.out.println("PASS: " + pass);
            System.out.println("FAIL: " + fail);

        } catch (Exception e) {
            System.out.println("Error reading dataset.");
        }

        scanner.close();
    }
}
